### PR TITLE
Fix: Jib push fails to push configured tag

### PR DESCRIPTION
**Pull request by dev-gaur**
fabric8:push with Jib Mode fails to push image tags with strings consisting "-SNAPSHOT".

### CODE REVIEW

File: core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java [hunk 1]
```diff
@@ -128,32 +130,41 @@ public static void jibPush(ImageConfiguration imageConfiguration, MavenProject p
...
-            }
+
         } catch (InvalidImageReferenceException | IllegalStateException e) {
             log.error("Exception occurred while pushing the image: %s", imageConfiguration.getName());
             throw new MojoExecutionException(e.getMessage(), e);
 
         }
     }
 
+    private static List<String> appendOriginalImageNameTagIfApplicable(List<String> tags, String imageName) {
```
Comment (manusa) [inline core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java]: Why not return a Set<String> ?

File: core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java [hunk 2]
```diff
@@ -128,32 +130,41 @@ public static void jibPush(ImageConfiguration imageConfiguration, MavenProject p
...
 
             Credential pushCredential = getRegistryCredentials(configuredRegistry, registryConfig);
             final List<String> tags = buildImageConfiguration.getTags();
-            if (tags.isEmpty()) {
-                final String targetImage = new ImageName(imageConfiguration.getName()).getFullName();
-                pushImage(baseImage, targetImage, pushCredential, log);
-            } else {
-                tags.stream().filter(Objects::nonNull).forEach(tag -> {
+
+            appendOriginalImageNameTagIfApplicable(tags, imageName).stream().filter(Objects::nonNull).forEach(tag -> {
```
Comment (manusa) [inline core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java]: Move the filter to the `appendOriginalImageNameTagIfApplicable` too

File: core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java [hunk 3]
```diff
@@ -128,32 +130,41 @@ public static void jibPush(ImageConfiguration imageConfiguration, MavenProject p
...
             log.error("Exception occurred while pushing the image: %s", imageConfiguration.getName());
             throw new MojoExecutionException(e.getMessage(), e);
 
         }
     }
 
+    private static List<String> appendOriginalImageNameTagIfApplicable(List<String> tags, String imageName) {
+
+        ImageName tempImage = new ImageName(imageName);
+        Set<String> tagSet = tags.stream().collect(Collectors.toSet());
```
Comment (manusa) [inline core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java]: Set the filter here instead.
```suggestion
        Set<String> tagSet = tags.stream().filter(Objects::nonNull).collect(Collectors.toSet());
```

File: core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java (hunk 1 above)
Comment (dev-gaur) [reply core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java]: aah that was my intent i forgot to change that

File: core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java (hunk 3 above)
Comment (dev-gaur) [reply core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java]: coolio

File: core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java [hunk 4]
```diff
@@ -128,32 +130,41 @@ public static void jibPush(ImageConfiguration imageConfiguration, MavenProject p
...
+
+        if (!tempImage.getTag().isEmpty()) {
+            // converting to set to ensure unique entries in the list
```
Comment (manusa) [inline core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java]: Also this comment is not necessary

**Pull request merged**